### ISSUE TITLE
Killed by iOS 12

### ISSUE THREAD

**Issue by finngaida** [eldade/EEIOKitListener#2]
Since iOS 10 already already pretty much turned off the tap on this data, iOS 12 killed it (almost) completely. :( 
This is what I got:

```
BatteryInstalled = 1;
ExternalConnected = 1;
IOClass = AppleARMPMUCharger;
```

Did anybody find a suitable alternative to get at least the Design Capacity?

**Comment by mice007** [comment]
Not yet. I'm trying to find any clues around the net but it seems we are screwed. However Battery Life app seems to do something, I'm following their app to see how they solve this.

**Comment by bill350** [comment]
Bad news.

@finngaida & @mice007 did you tried to use any of this IOKit code in production? Did it pass the Apple Review team?

**Comment by mice007** [comment]
@bill350 yes. I've implemented a code that worked fine until iOS 12 but around Apple introduced their battery health page the review team asked me to get rid of the battery health code on the upcoming submit to avoid a permanent ban.

**Comment by finngaida** [comment]
Uf @mice007 my condolences. That makes it even worse. Lets wait for iOS 12 GM or final and see if anything changes

**Comment by bill350** [comment]
Thanks for those pieces of information @mice007 !
If the review team asked you to get rid of the battery health code, it not explains why apps like [Battery Life](https://itunes.apple.com/fr/app/battery-life/id1080930585?mt=8) are still on the AppStore 🤔

**Comment by mice007** [comment]
@bill350 I had the same question. Probably they using something similar to read battery information. And as I saw they have the same issues under iOS 12, I'm curious about they will find a solution or not. Also it is possible that they got a similar warning but modified the code before a new submit and risked to get banned.

**Comment by bill350** [comment]
@mice007 @eldade Two weeks ago, Battery Life passed the Apple Review team validation for their update.

**Comment by emichel11** [comment]
Hi, 
I tried with IOS 13.2 - the takes throws an exception. Before I investigate on this - is there any chance to get battery information? AT least I need battery temperature an charge in % ?

**Comment by jonasman** [comment]
Hi guys, even tho it is not possible to send to the store, is there any private api that we can use to check charging power? Im interested in having an app to tell me this info that i wont publish
